plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -51,7 +51,7 @@ simplify = DeadCode.removeDeadBindings . Inline.inline . Beta.beta
 
 -- | Perform some simplification of a 'Term'.
 simplifyTerm :: Compiling m e uni fun a => Term TyName Name uni fun b -> m (Term TyName Name uni fun b)
-simplifyTerm = PLC.rename >=> (runIfOpts $ pure . simplify)
+simplifyTerm = PLC.rename >=> (runIfOpts $ pure . simplify . simplify . simplify . simplify)
```

Comment:
Maybe let's add a "simplifier iterations" option so we can easily tweak how often we do this?

Reply:
TBH we might want to just run to fixpoint!

Reply:
Although that'll probably require us to change our passes to return a flag saying whether they changed anything, which would be a bit of work.

Reply:
Added an argument to simplify to specify the number of iterations to the simplifier.